common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java

```diff
@@ -0,0 +1,48 @@
+package com.gradle;
+
+import com.gradle.maven.extension.api.scan.BuildScanApi;
+import groovy.lang.Binding;
+import groovy.lang.GroovyShell;
+import org.apache.maven.MavenExecutionException;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.monitor.logging.DefaultLog;
+import org.codehaus.plexus.logging.Logger;
+
+import java.io.File;
+import java.io.IOException;
+
+class GroovyScriptUserData {
+
+    static void addToBuildScan(MavenSession session, BuildScanApi buildScan, Logger logger) throws MavenExecutionException {
+        File scriptFile = getScriptFile(session);
+        if (scriptFile.exists()) {
+            logger.debug("Evaluating custom user data Groovy script: " + scriptFile);
+            evaluateGroovyScript(session, buildScan, logger, scriptFile);
+        } else {
+            logger.debug("Skipping evaluation of custom user data Groovy script because it does not exist: " + scriptFile);
+        }
+    }
+
+    private static File getScriptFile(MavenSession session) {
+        File rootDir = session.getRequest().getMultiModuleProjectDirectory();
+        return new File(rootDir, ".mvn/gradle-enterprise-custom-user-data.groovy");
```

Comment:
❓ I guess it's enough when the script filename is fixed. But where/how will we document this to the user?

Reply:
We should probably add it [to the docs](https://docs.gradle.com/enterprise/maven-extension/#custom_extension) and the trial navigator.

Reply:
+1 I don't think we need to offer customization of this file